source-academy/frontend
# Add Publish Grading Feature

### Description

(closes #2522)



Currently, the grading states design and business logic cause ambiguity with resubmission of assessments being assumed to be graded immediately, and with grades being displayed when they should not.

This feature aims to solve this issue by introducing a new grading state `published` and a new backend boolean `is_grading_published`, allowing explicit publishing of student's grades on the frontend and preventing the fetching of grades if they have yet to be published, hence creating a better logic flow.

This pull request contains changes to be made to the frontend to enable this new feature.

### Type of change


### How to test


(Import your own assessment XML files for testing if possible)

1. Using a test student account, submit an assessment
2. Using a test staff/admin account, grade some but not all questions in the assessment
3. Staff/admin should not be able to publish grades if not all questions have been graded
4. Using a test staff/admin account, complete grading the assessment
5. The grades should not be visible to the student even upon all questions being graded
6. Using a test staff/admin account, publish the assessment grades using the new UI button in the grading dashboard
7. The grades should now be visible to the student
8. Using a test staff/admin account, unsubmit the assessment
9. The assessment should be unpublished automatically and the student should not be able to view any XP for that assessment, even if the student resubmits again
10. Students should still be able to view comments
11. Once regrading is done, publish grades again and the student should be able to view the grades
12. Staff/Admin is also able to unpublish the grades after publishing them if required

### Checklist
